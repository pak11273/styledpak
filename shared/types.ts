import {
  AlignItemsProps,
  AlignSelfProps,
  BorderRadiusProps,
  ButtonStyleProps,
  ColorProps,
  DisplayProps,
  FlexboxProps,
  FontSizeProps,
  GridProps,
  HeightProps,
  LayoutProps,
  LineHeightProps,
  MaxHeightProps,
  MaxWidthProps,
  MinHeightProps,
  MinWidthProps,
  OrderProps,
  OverflowProps,
  SpaceProps,
  TypographyProps,
  WidthProps,
} from "styled-system"

export type SystemProps = AlignSelfProps &
  AlignItemsProps &
  AlignSelfProps &
  ButtonStyleProps &
  BorderRadiusProps &
  ColorProps &
  DisplayProps &
  FlexboxProps &
  FontSizeProps &
  GridProps &
  HeightProps &
  LayoutProps &
  LineHeightProps &
  MaxHeightProps &
  MaxWidthProps &
  MinHeightProps &
  MinWidthProps &
  OrderProps &
  OverflowProps &
  SpaceProps &
  TypographyProps &
  WidthProps
