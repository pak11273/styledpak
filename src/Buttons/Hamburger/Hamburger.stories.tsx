import { Hamburger } from "./Hamburger"
import React from "react"

export default {
  title: "Hamburger",
}

export const Base = () => {
  const someParentFunc = () => {
    return "blah"
  }

  return (
    <div
      style={{
        display: "flex",
        justifyContent: "center",
        position: "relative",
      }}
    >
      <Hamburger bg="blue" size="small" variant="secondary" />
    </div>
  )
}
