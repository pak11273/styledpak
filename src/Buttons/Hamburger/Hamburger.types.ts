export interface HamburgerProps {
  open?: string
  onClick?: () => void
}
