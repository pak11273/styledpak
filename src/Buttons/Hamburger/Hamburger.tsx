import React, { useState } from "react"
import { color, layout } from "styled-system"

import { Button } from "../../index"
import { HamburgerProps } from "./Hamburger.types"
import { SystemProps } from "../../../shared/types"
import styled from "styled-components/"

const Styles = styled(Button)<SystemProps>`
  top: 5%;
  left: 2rem;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;

  &:focus {
    outline: none;
  }

  div {
    width: 2rem;
    height: 0.25rem;
    background: ${({ open }) => (open ? "#0D0C1D" : "#EFFFFA")};
    border-radius: 10px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;

    :first-child {
      transform: ${({ open }) => (open ? "rotate(45deg)" : "rotate(0)")};
    }

    :nth-child(2) {
      opacity: ${({ open }) => (open ? "0" : "1")};
      transform: ${({ open }) => (open ? "translateX(20px)" : "translateX(0)")};
    }

    :nth-child(3) {
      transform: ${({ open }) => (open ? "rotate(-45deg)" : "rotate(0)")};
    }
  }
  ${color};
  ${layout};
`

export const Hamburger = (props: any) => {
  const [open, setOpen] = useState(false)
  return (
    <Styles {...props} onClick={() => alert("hi")}>
      <div />
      <div />
      <div />
    </Styles>
  )
}
