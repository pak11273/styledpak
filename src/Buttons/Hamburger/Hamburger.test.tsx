import { Hamburger } from "./Hamburger"
import { HamburgerProps } from "./Hamburger.types"
import React from "react"
import { render } from "@testing-library/react"

describe("Test Component", () => {
  let props: HamburgerProps

  const renderComponent = () => render(<Hamburger {...props} />)

  it("should render as a default button", () => {
    expect(renderComponent).toMatchSnapshot()
  })
})
