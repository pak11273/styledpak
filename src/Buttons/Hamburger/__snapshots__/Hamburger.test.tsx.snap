// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component should render as a default button 1`] = `[Function]`;
